doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,339 @@
+    }
+  ]
+}
+```
+#### 3.3 Porting guide
+1. git clone sonic-platform-common to get the S3IP framework
+2. Verify the availability of the S3IP framework on the host computer
+ - Generate a host package, run the following command
+     - dpkg-buildpackage -rfakeroot -b -US -UC
+ - Install the package, run the following command
+     - dpkg -i s3ip-sysfs_1.0.0_amd64.deb
+ - Check the /sys_switch directory, run the following command
+     - tree -psv /sys_switch/
+3. Porting S3IP sysfs framework to platform project
+ - Create the S3IP sysfs service configuration file. 
+     - Modify /s3ip/s3ip_sysfs_service/s3ip_sysfs_conf.json to create a configuration file
+     - This file should be installed to /etc/s3ip/s3ip_sysfs_conf.json on the device
+ - Implement the platform hardware driver, and export parameters through s3ip_sysfs.ko
+     - s3ip_sysfs. ko is generated by compiling /s3ip/s3ip_sysfs_frame source code
+ - Install the s3ip-sysfs.service on the device
+    - Copy service file /s3ip/scripts/s3ip-sysfs.service to the project.
+    - The service file needs to be installed in /etc/systemd/system/s3ip-sysfs.service on the device.
+    - When the platform package is installed, the vendor should enable s3ip-sysfs.service and start this service automatically when the device reboot
+    - After the platform package is uninstalled, the vendor should disable the s3ip-sysfs.service and remove the service file
+4. Self-test after integration
+   - Ensure that the /sys_switch directory conforms to the S3IP sysfs specification 
+   - Hardware property information can be read and written normally
+
+### 4 Compare with [PDDF](https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md)
+
+The S3IP sysfs specification is intended to provide a more general set of hardware access interfaces. Using sysfs is definitely the best option, as it fits into the Linux design philosophy of everything being a file, and there are more tools and languages to deal with it.
+
+For traditional driver engineers, The S3IP SYSFS interface is very friendly and easy to debug, they only need to care about the content and format of the sysfs node. So the learning cost is zero and it's easy to understand and implement. 
+
+For ODM vendors new to SONIC, opt for PDDF because you can do platform adaptation with a configuration file, which is cool and popular.
+
+S3IP SYSFS and PDDF both share similar goals of reducing vendor adaptation effort and increasing the reusability of device management software.
+
+However, there are many differences between the two frameworks, which are listed below. ODM vendors can choose the framework based on actual requirements.
+
+| Item  | S3IP sysfs|PDDF|
+|-|-|-|
+|Requirements| The requirements are put forward from the user's perspective, and the sysfs node is summarized from the actual operation experience. The goal is to unify the interface for device management | The requirements are proposed from the technical point of view.TBD

```

Comment:
*** Requirements: (PDDF) The requirements are proposed from technical point of view aimed at platform driver and APIs development in SONiC. Only those SysFS which are required by SONiC platform APIs are exposed.

Reply:
Please add this comment in the PDDF column for the row "Requirements"

Reply:
already updated